deepchem/feat/vocabulary_builders/vocabulary_builder.py

```diff
@@ -0,0 +1,48 @@
+from deepchem.data import Dataset
+
+
+class VocabularyBuilder():
```

Comment:
I think this basic API looks reasonable

Reply:
I will break this into 4 subsequent pull requests and add them one at a time - base vocabulary builder (#3265), hugging face tokenizer, hugging face vocabulary builder and lastly, grover.